### PR TITLE
📖 Documentation for branching

### PR DESCRIPTION
**Pull request by Enriqe**
The following PR follows what was written by @bramanudom in #21614 but cuts down to the current capabilities introduced by #25502.

#20083

### CODE REVIEW

File: extensions/amp-story/amp-story.md [hunk 1]
```diff
@@ -785,6 +785,62 @@ You can apply multiple entrance animations on one element (for example, an eleme
...
+
+With branching, AMP Stories now supports URLs in the form of:
+
+```
+https://www.mydomain.com/good-story/#page=<page-id>
+```
+
+where `page-id` refers to the unique id of an `amp-story-page`. You can also use the fragment parameter and the `page-id` value like an anchor link in some use cases. See [Integration with Sidebar for Stories](#integration-with-sidebar-for-stories) for an example.
+
+### Sharing a Specific Story Page
```
Comment (gmajoulet) [inline extensions/amp-story/amp-story.md]: This part has been removed from the code

File: extensions/amp-story/amp-story.md [hunk 2]
```diff
@@ -785,6 +785,62 @@ You can apply multiple entrance animations on one element (for example, an eleme
...
 
+## Branching
+
+Stories support fragment paramenters in the url. By utilizing this, you can create stories that start from somewhere other than the beginning, share specific pages of stories, or even allow users to jump around within a story. An example is a table of contents.
+
+### Manipulating Navigation
+
+The feature allows you to manipulate navigation within a story. When using it, navigation throughout the story from a 'branch-off point' will follow the path that a user takes. If a user navigates from `page-1` to `page-5` and then `page-6`, navigating backwards will follow this exact path (`page-6` to `page-5` and finally `page-1`). The other pages will not be exposed to the user.
+
+### Sharing Options
```
Comment (gmajoulet) [inline extensions/amp-story/amp-story.md]: This part has been removed from the code

File: extensions/amp-story/amp-story.md [hunk 3]
```diff
@@ -785,6 +785,62 @@ You can apply multiple entrance animations on one element (for example, an eleme
 If a composed animation is supposed to start after the end of a separate element's animation, make sure that all nested elements that compose the animation have the attribute `animate-in-after` set to the same `id`.
 {% endcall %}
 
+## Branching
+
+Stories support fragment paramenters in the url. By utilizing this, you can create stories that start from somewhere other than the beginning, share specific pages of stories, or even allow users to jump around within a story. An example is a table of contents.
+
+### Manipulating Navigation
```
Comment (gmajoulet) [inline extensions/amp-story/amp-story.md]: Given the modifications we made to the feature, I think this should go last, as the two interesting things are the sidebar and the URL fragment parameter

File: extensions/amp-story/amp-story.md (hunk 2 above)
Comment (Enriqe) [reply extensions/amp-story/amp-story.md]: Can't users share specific pages through the URLs via the fragment params?

File: extensions/amp-story/amp-story.md (hunk 1 above)
Comment (Enriqe) [reply extensions/amp-story/amp-story.md]: Ditto last comment, not sure if I'm missing something.

File: extensions/amp-story/amp-story.md [hunk 4]
```diff
@@ -785,6 +785,62 @@ You can apply multiple entrance animations on one element (for example, an eleme
 If a composed animation is supposed to start after the end of a separate element's animation, make sure that all nested elements that compose the animation have the attribute `animate-in-after` set to the same `id`.
 {% endcall %}
 
+## Branching
+
+Stories support fragment paramenters in the url. By utilizing this, you can create stories that start from somewhere other than the beginning, share specific pages of stories, or even allow users to jump around within a story. An example is a table of contents.
```
Comment (CrystalOnScript) [inline extensions/amp-story/amp-story.md]: "paramenters" --> "parameters"

Comment (CrystalOnScript) [reply extensions/amp-story/amp-story.md]: Id rephrase to:

Branching enables the identification of individual story pages. Users can jump around within a story, start a story from somewhere other than the beginning, and share specific story pages.  An example is a table of contents or multiple choice buttons. 

Fragment parameters in the url supports this feature.

File: extensions/amp-story/amp-story.md [hunk 5]
```diff
@@ -785,6 +785,62 @@ You can apply multiple entrance animations on one element (for example, an eleme
...
+
+### Manipulating Navigation
+
+The feature allows you to manipulate navigation within a story. When using it, navigation throughout the story from a 'branch-off point' will follow the path that a user takes. If a user navigates from `page-1` to `page-5` and then `page-6`, navigating backwards will follow this exact path (`page-6` to `page-5` and finally `page-1`). The other pages will not be exposed to the user.
```
Comment (CrystalOnScript) [inline extensions/amp-story/amp-story.md]: I'd update to:

Branching allows navigation manipulate within a story. The story tracks navigation. If a user navigates from `page-1` to `page-5` and then `page-6`, navigating backwards will follow the exact path. The skipped pages are not exposed to the user when navigating backwards, it will follow `page-6`, to `page-5`, and ends back at `page-1`.

File: extensions/amp-story/amp-story.md [hunk 6]
```diff
@@ -785,6 +785,62 @@ You can apply multiple entrance animations on one element (for example, an eleme
...
+
+### Sharing Options
+
+Specific pages within a story are shareable through URLs.
```
Comment (CrystalOnScript) [inline extensions/amp-story/amp-story.md]: Some uses of `url` are lowercase and other are capitalized. Can we please capitalize all of them?

Comment (CrystalOnScript) [reply extensions/amp-story/amp-story.md]: I'd rephrase to:
User's can share a specific page within a story through URLs.

Comment (CrystalOnScript) [reply extensions/amp-story/amp-story.md]: Will we be providing a share button for this feature?

File: extensions/amp-story/amp-story.md [hunk 7]
```diff
@@ -785,6 +785,62 @@ You can apply multiple entrance animations on one element (for example, an eleme
...
+
+```
+https://www.mydomain.com/good-story/#page=<page-id>
+```
+
+where `page-id` refers to the unique id of an `amp-story-page`. You can also use the fragment parameter and the `page-id` value like an anchor link in some use cases. See [Integration with Sidebar for Stories](#integration-with-sidebar-for-stories) for an example.
+
+### Sharing a Specific Story Page
+
+With the branching feature, users will have the option to share the story from the page they are currently seeing. If a user selects the option to share from the page they are currently seeing, the URL generated will be in the form described above.
```
Comment (CrystalOnScript) [inline extensions/amp-story/amp-story.md]: Well, if it hasn't been removed, I'd rephrase as:

Branching grants users the option to share the the story page they are currently seeing. URLs are generated when the user selects the option to share. Sending the URL will direct the receiving user to that exact story page, not the beginning of the story. 

Also, I'd update the order. I'd place the URL Fragment Parameter section underneath this one and remove Sharing options, as the same information is covered here.

File: extensions/amp-story/amp-story.md [hunk 8]
```diff
@@ -785,6 +785,62 @@ You can apply multiple entrance animations on one element (for example, an eleme
...
+
+[tip type="important"]
+**Important**, this may expose the `amp-story-page` ids to the user.
+[/tip]
+
+## Integration with Sidebar for Stories
+
+The use of `amp-sidebar` is supported for use within `amp-story`. See the [Sidebar for Stories documentation](https://amp.dev/documentation/components/amp-sidebar?format=websites#sidebar-for-stories) for information on limitations and caveats.
```
Comment (CrystalOnScript) [inline extensions/amp-story/amp-story.md]: I'd rephrase to

`amp-story` supports the use of `amp-sidebar` with a few limitations and caveats. See the Sidebar for Stories documentation for more details.

File: extensions/amp-story/amp-story.md [hunk 9]
```diff
@@ -785,6 +785,62 @@ You can apply multiple entrance animations on one element (for example, an eleme
...
+
+By using both branching features and `amp-sidebar`, you can create stories that have a table of contents. To do this, make use of URL fragment parameter.
```
Comment (CrystalOnScript) [inline extensions/amp-story/amp-story.md]: I'm sorry, this doesn't make sense to me. How do I make use of the URL fragment parameter? I thought it was auto generate?

File: extensions/amp-story/amp-story.md [hunk 10]
```diff
@@ -785,6 +785,62 @@ You can apply multiple entrance animations on one element (for example, an eleme
...
+
+An example follows. We add an `amp-sidebar` that has in it both a table of contents and out-links to other links of interest.
```
Comment (CrystalOnScript) [inline extensions/amp-story/amp-story.md]: I'd rephrase to:

The following example demonstrates a table of contents inside of an `amp-sidebar`. The table of contents has a link to a specific story page, and and out-link to a different website.

File: extensions/amp-story/amp-story.md (hunk 5 above)
Comment (Enriqe) [reply extensions/amp-story/amp-story.md]: Done. I changed your manipulate to manipulation because I think this is what you meant, but let me know if I'm wrong :)

File: extensions/amp-story/amp-story.md (hunk 6 above)
Comment (Enriqe) [reply extensions/amp-story/amp-story.md]: I don't think so. I just understood what Gabriel was saying, and that we are not actually going to be generating the URLs as the user navigate through the story. 

But the publisher (or even a user if they know the page ID) could share the story starting on a specific page by appending the fragment parameter in the URL

File: extensions/amp-story/amp-story.md (hunk 8 above)
Comment (Enriqe) [reply extensions/amp-story/amp-story.md]: Done.

File: extensions/amp-story/amp-story.md (hunk 9 above)
Comment (Enriqe) [reply extensions/amp-story/amp-story.md]: Yeah it's a bit confusing because we split the feature and removed some of its capabilities. But basically right now we don't generate the URL automatically, you would have to append the fragment parameter in the URL with the page id manually.

**Comment by Enriqe** [comment]
Wohoo thanks for the reviews!
